### PR TITLE
Npm test

### LINKED ISSUE #4: Write npm test
To avoid issue like this #1, there should be npm test

### PR DESCRIPTION
**Pull request by pascalgrimaud**
Fix #4

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -29,7 +29,7 @@
   });
 
   gulp.task('pre-test', function () {
-    return gulp.src('generators/**/*.js')
+    return gulp.src('generators/app/*.js')
```
Comment (pascalgrimaud) [inline gulpfile.js]: Not really sure about this modification

Comment (deepu105) [inline gulpfile.js]: Both means the same for current file structure, your change makes it even more strict which is fine

File: package.json [hunk 2]
```diff
@@ -39,7 +40,12 @@
       "gulp-rename": "^1.2.0",
       "gulp-sequence": "0.4.4",
       "gulp-shell": "^0.5.1",
-      "yeoman-assert": "2.1.1"
+      "mocha": "2.3.4",
+      "yeoman-assert": "2.1.1",
+      "yeoman-test": "1.0.0"
+    },
+    "scripts": {
+        "test": "mocha test/*"
```
Comment (deepu105) [inline package.json]: I thing there is already a test task in gulp file as well rite?

File: gulpfile.js (hunk 1 above)
Comment (pascalgrimaud) [inline gulpfile.js]: In fact, if I use **, I got this error :
```bash
events.js:141
      throw er; // Unhandled 'error' event
      ^
Error: Unable to parse /home/pgrimaud/workspace/generator-jhipster-module/generators/app/templates/generators/app/_index.js

Line 67: Unexpected token %
```

File: package.json (hunk 2 above)
Comment (pascalgrimaud) [inline package.json]: Yes, it's in gulpfile.js, but by adding in package.json, tests can be launch with `npm test` too
- package.json => npm test
- gulpfile.js => gulp test
Should we keep both ?

Comment (deepu105) [inline package.json]: I think should be ok to keep in both

**Comment by pascalgrimaud** [comment]
so let's merge!

**Pull request merged**